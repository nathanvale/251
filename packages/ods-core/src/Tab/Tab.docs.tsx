import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { IconElectricity } from "@origin-digital/ods-icons";
import { Tab } from "./Tab";

export const docs: ComponentDocs = {
  category: "Content",
  componentName: "Tab",
  description: `An individual Tab, only useful as the child of the Tabs component.`,
  migrationGuide: false,
  propDescriptions: {
    value: "This tab is selected when this value matches the parent Tabs value",
    label: "Text label for the tab",
    icon: "Optional icon to be displayed to the left of the label",
    children:
      "The content to be shown when the tab is selected (handled by the Tabs component)",
    muiProps: "Allows access to the MUI component API under the hood",
  },
  examples: {
    default: {
      Code: () => <Tab data-id="tab1" label="Electricity" />,
    },
    additional: [
      {
        label: "Icon tab",
        description: "Tab with an icon on it",
        Code: () => (
          <Tab
            data-id="tab1"
            label="Electricity"
            icon={<IconElectricity tone="neutral" />}
          />
        ),
      },
    ],
  },
  snippets: [],
};
