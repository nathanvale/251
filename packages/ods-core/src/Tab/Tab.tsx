import React from "react";
import {
  Tab as MuiTab,
  TabProps as MuiTabProps,
  makeStyles,
  Theme,
  fade,
} from "@material-ui/core";
import { ComponentBaseProps } from "@origin-digital/ods-types";
import { useTracking } from "../TrackingProvider/useTracking";
import { Box } from "../Box";

export interface TabProps extends ComponentBaseProps {
  label: string | React.ReactNode;
  icon?: React.ReactNode;
  value?: string | number;
  muiProps?: MuiTabProps;
}

const useTabStyles = makeStyles(
  (theme: Theme) => ({
    root: {
      color: theme.palette.secondaryB.main,
      opacity: 1,
      textTransform: "none",
      padding: `${theme.spacing(3)}px ${theme.spacing(4)}px`,
      fontSize: `${theme.typography.body1.fontSize}px`,
      "&:hover": {
        color: theme.palette.grey[600],
        backgroundColor: theme.palette.grey[100],
      },
      "&$selected .MuiTab-wrapper": {
        fontWeight: theme.typography.fontWeightMedium,
      },
      "&$selected svg": {
        fill: theme.palette.secondary.main,
        color: theme.palette.secondary.main,
      },
      "&$selected:hover": {
        backgroundColor: fade(theme.palette.secondary.main, 0.08),
      },
      "&:focus": {
        color: theme.palette.grey[600],
        backgroundColor: fade(theme.palette.secondary.main, 0.16),
      },
      "&:active": {
        color: theme.palette.grey[600],
        backgroundColor: fade(theme.palette.secondary.main, 0.24),
      },
    },
    selected: {},
  }),
  { classNamePrefix: "tab" }
);

export const Tab = ({
  icon,
  label,
  "data-id": dataId = "tab",
  muiProps,
  value,
  ...others
}: TabProps) => {
  const classes = useTabStyles();
  const { onClickCapture, ref } = useTracking({
    children: label,
    "data-id": dataId,
    type: Tab.displayName,
    postClickState: "selected",
  });
  return (
    <MuiTab
      disableRipple
      data-id={dataId}
      classes={classes}
      onClickCapture={onClickCapture}
      ref={ref}
      label={
        <Box display="flex" flexDirection="row" justifyContent="center">
          {icon && <Box marginRight="xsmall">{icon}</Box>}
          {label}
        </Box>
      }
      value={value}
      id={`tab-${value}`}
      aria-controls={`tabpanel-${value}`}
      {...muiProps}
      {...others}
    />
  );
};

Tab.defaultProps = {
  "data-id": "tab",
};
Tab.displayName = "Tab";
