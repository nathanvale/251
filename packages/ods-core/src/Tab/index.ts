export { Tab } from "./Tab";
export type TabProps = import("./Tab").TabProps;
