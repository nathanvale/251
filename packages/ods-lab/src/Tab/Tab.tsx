import React from "react";
import MuiTab, {TabProps as ITab} from "@material-ui/core/Tab";

export interface TabProps extends ITab {}

export const Tab = (props: TabProps) => (
  <MuiTab {...props}>{props.children}</MuiTab>
);
