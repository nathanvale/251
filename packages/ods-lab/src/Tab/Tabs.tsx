import React from "react";
import MuiTabs, {TabsProps as ITabs} from "@material-ui/core/Tabs";

export interface TabsProps extends ITabs {}

export const Tabs = (props: TabsProps) => (
  <MuiTabs {...props} TabIndicatorProps={{children: <div />}}>
    {props.children}
  </MuiTabs>
);
