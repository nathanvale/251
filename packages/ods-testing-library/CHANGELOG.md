# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
