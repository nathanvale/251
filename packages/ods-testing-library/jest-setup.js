import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
