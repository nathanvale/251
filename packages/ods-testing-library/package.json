{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
