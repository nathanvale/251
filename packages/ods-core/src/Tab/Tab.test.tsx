import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Tab } from "../Tab";

test("it renders", () => {
  const { container } = render(<Tab label="Melbourne" id="mlb" value="mlb" />);
  expect(container).toMatchSnapshot();
});
