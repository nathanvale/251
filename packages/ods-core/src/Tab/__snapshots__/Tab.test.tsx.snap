// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <button
    aria-controls="tabpanel-mlb"
    class="MuiButtonBase-root MuiTab-root tab-root-1 MuiTab-textColorInherit"
    data-id="tab"
    id="mlb"
    role="tab"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTab-wrapper"
    >
      <div
        class="c0"
        display="flex"
      >
        Melbourne
      </div>
    </span>
  </button>
</div>
`;
